Reducers don't seem to work in templated functions

### Describe the bug
If I use a reducer in a templated function the compiler gives me an error

### Expected behavior

The reducer to work

### OpenCilk version

clang version 14.0.6 (https://github.com/OpenCilk/opencilk-project fc90ded2b090672f84c58d12d8d85cd999eb6c1a)
Target: x86_64-unknown-linux-gnu
Thread model: posix


### Steps to reproduce (include relevant output)

compile the following code 

```
#include <cilk/cilk.h>

void zero(void *v) { *(int *)v = 0; }

void plus(void *l, void *r) { *(int *)l += *(int *)r; }

template <typename T> int f() {
  int cilk_reducer(zero, plus) red;
  red += 5;
  return red;
}

int main() { return f<int>(); }
```

with `clang++   -fopencilk -o basic test.cpp`


The error is 
```
warning: ISO C++ does not allow indirection on operand of type 'void *' [-Wvoid-ptr-dereference]
test.cpp:13:21: note: in instantiation of function template specialization 'f<int>' requested here
int main() { return f<int>(); }
                    ^
test.cpp:9:7: error: invalid operands to binary expression ('void' and 'int')
  red += 5;
      ^  ~
warning: ISO C++ does not allow indirection on operand of type 'void *' [-Wvoid-ptr-dereference]
error: cannot initialize return object of type 'int' with an lvalue of type 'void'
2 warnings and 2 errors generated.
```


### Additional comments

If I just remove the templates it compiles and seems to work.

--- a/clang/lib/Sema/SemaExpr.cpp
+++ b/clang/lib/Sema/SemaExpr.cpp
@@ -2029,7 +2029,7 @@ Expr *Sema::BuildHyperobjectLookup(Expr *E, bool Pointer) {
   if (!HT)
     return E;
 
-  bool Difficult = InputType->isDependentType();
+  bool Difficult = CurContext->isDependentContext();
 
   // For now all hyperobjects use the same lookup function.
   IdentifierInfo *ID = PP.getIdentifierInfo("__hyper_lookup");

--- a/clang/test/Cilk/hyper-expand3.cpp
+++ b/clang/test/Cilk/hyper-expand3.cpp
@@ -1,0 +1,17 @@
+// RUN: %clang_cc1 %s -x c++ -O1 -fopencilk -verify -S -fsyntax-only
+// RUN: %clang_cc1 --std=c++17 %s -x c++ -O1 -fopencilk -verify -S -fsyntax-only
+// expected-no-diagnostics
+// See opencilk-project issue 132.  The reference to hyperobject red
+// does not have a dependent type but arises in a dependent context.
+
+void zero(void *v) { *(int *)v = 0; }
+
+void plus(void *l, void *r) { *(int *)l += *(int *)r; }
+
+template <typename T> int f() {
+  int _Hyperobject(zero, plus) red;
+  red += 5;
+  return red;
+}
+
+int main() { return f<int>(); }

